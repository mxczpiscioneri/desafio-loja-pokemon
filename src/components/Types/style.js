import styled from 'styled-components'
import { Avatar, Box, Chip } from '@material-ui/core'

export const TypesStyled = styled(Box)`
  display: flex;
  justify-content: space-evenly;
  margin: 8px 0;
`
export const ChipStyled = styled(Chip)`
  min-width: 32px;
  
  div {
    margin: ${props => (props.colapse && '0 !important')};
  }
  
  span {
    padding: ${props => (props.colapse ? 0 : '12px')};
  }
`
export const AvatarStyled = styled(Avatar)`
  background-color: ${props => props.color};
  padding: 4px;
`
