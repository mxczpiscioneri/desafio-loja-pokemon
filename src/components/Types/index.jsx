import React from 'react'
import PropTypes from 'prop-types'
import { getType } from '../../common/constants/typesOfPokemon'
import { AvatarStyled, ChipStyled, TypesStyled } from './style'

const Types = ({ data, colapse }) => (
  <TypesStyled>
    {data.map(type => (
      <ChipStyled
        key={type.type.name}
        label={!colapse && type.type.name}
        colapse={colapse ? 1 : 0}
        avatar={
          <AvatarStyled
            color={getType(type.type.name).color}
            src={require(`../../assets/icons/${type.type.name}.svg`).default}
            alt={type.type.name}
            title={type.type.name} />
        }
        variant="outlined"
      />
    ))}
  </TypesStyled>
)

Types.propTypes = {
  data: PropTypes.array.isRequired,
  colapse: PropTypes.bool.isRequired
}

export default Types
