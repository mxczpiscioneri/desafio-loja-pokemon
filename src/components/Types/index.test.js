import React from 'react'
import { render } from '@testing-library/react'
import Types from '.'
import mockDetailsPokemon from '../../assets/mock-data/details-pokemon.json'

describe('Types components', () => {
  test('Render with success', () => {
    const component = render(<Types data={mockDetailsPokemon.types} colapse={false} />)
    expect(component).toMatchSnapshot()
  })

  test('Render colapse with success', () => {
    const component = render(<Types data={mockDetailsPokemon.types} colapse={true} />)
    expect(component).toMatchSnapshot()
  })
})
