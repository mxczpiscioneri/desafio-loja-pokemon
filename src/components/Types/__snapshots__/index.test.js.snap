// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Types components Render colapse with success 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-2 sc-bdfBwQ jCnDFQ"
      >
        <div
          class="MuiChip-root sc-gsTCUz fursvC MuiChip-outlined"
          colapse="1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm bcBWcq MuiChip-avatar"
            color="#78c851"
            title="grass"
          >
            <img
              alt="grass"
              class="MuiAvatar-img"
              src="grass.svg"
            />
          </div>
          <span
            class="MuiChip-label"
          />
        </div>
        <div
          class="MuiChip-root sc-gsTCUz fursvC MuiChip-outlined"
          colapse="1"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm iREEXP MuiChip-avatar"
            color="#a04aa0"
            title="poison"
          >
            <img
              alt="poison"
              class="MuiAvatar-img"
              src="poison.svg"
            />
          </div>
          <span
            class="MuiChip-label"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-2 sc-bdfBwQ jCnDFQ"
    >
      <div
        class="MuiChip-root sc-gsTCUz fursvC MuiChip-outlined"
        colapse="1"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm bcBWcq MuiChip-avatar"
          color="#78c851"
          title="grass"
        >
          <img
            alt="grass"
            class="MuiAvatar-img"
            src="grass.svg"
          />
        </div>
        <span
          class="MuiChip-label"
        />
      </div>
      <div
        class="MuiChip-root sc-gsTCUz fursvC MuiChip-outlined"
        colapse="1"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm iREEXP MuiChip-avatar"
          color="#a04aa0"
          title="poison"
        >
          <img
            alt="poison"
            class="MuiAvatar-img"
            src="poison.svg"
          />
        </div>
        <span
          class="MuiChip-label"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Types components Render with success 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1 sc-bdfBwQ jCnDFQ"
      >
        <div
          class="MuiChip-root sc-gsTCUz kKUUUe MuiChip-outlined"
          colapse="0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm bcBWcq MuiChip-avatar"
            color="#78c851"
            title="grass"
          >
            <img
              alt="grass"
              class="MuiAvatar-img"
              src="grass.svg"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            grass
          </span>
        </div>
        <div
          class="MuiChip-root sc-gsTCUz kKUUUe MuiChip-outlined"
          colapse="0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm iREEXP MuiChip-avatar"
            color="#a04aa0"
            title="poison"
          >
            <img
              alt="poison"
              class="MuiAvatar-img"
              src="poison.svg"
            />
          </div>
          <span
            class="MuiChip-label"
          >
            poison
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1 sc-bdfBwQ jCnDFQ"
    >
      <div
        class="MuiChip-root sc-gsTCUz kKUUUe MuiChip-outlined"
        colapse="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm bcBWcq MuiChip-avatar"
          color="#78c851"
          title="grass"
        >
          <img
            alt="grass"
            class="MuiAvatar-img"
            src="grass.svg"
          />
        </div>
        <span
          class="MuiChip-label"
        >
          grass
        </span>
      </div>
      <div
        class="MuiChip-root sc-gsTCUz kKUUUe MuiChip-outlined"
        colapse="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle sc-dlfnbm iREEXP MuiChip-avatar"
          color="#a04aa0"
          title="poison"
        >
          <img
            alt="poison"
            class="MuiAvatar-img"
            src="poison.svg"
          />
        </div>
        <span
          class="MuiChip-label"
        >
          poison
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
